v1.16-4.8/Dockerfile

```diff
@@ -39,6 +40,7 @@ RUN apk add --no-cache \
          google-protobuf:3.21.12 \
          nio4r:2.7.3 \
          snappy:0.3.0 \
+         rdkafka:0.16.0 \
```

Comment:
Thank you for your contribution @onecer, please add `rdkafka` to the `outputs/Gemfile` and update the `Gemfile.lock` as well.

Reply:
Okay

Reply:
> Thank you for your contribution @onecer, please add `rdkafka` to the `outputs/Gemfile` and update the `Gemfile.lock` as well.

We cannot do it that way. This plugin required `make`  to build it. It is a c extension. 

```
make DESTDIR\= sitearchdir\=./.gem.20240710-7-kjbl0d sitelibdir\=./.gem.20240710-7-kjbl0d
make failedNo such file or directory - make
```

In Dockerfile build stage, before we start install Gemfile.outputs. `make` have been deleted.

```
 && apk del $BUILD_DEPS \
```

Reply:
we should add make to that build step again and remove it at the end. the goal is to have all the non-essential output plugins added in the last image layer only

Reply:
> we should add make to that build step again and remove it at the end. the goal is to have all the non-essential output plugins added in the last image layer only

Done.

I use this to make an image and update the lock file.


Dockerfile-base
```
# TODO every time the base image is changed please make sure BASEIMAGEDEPS is updated (further down in this file)
# https://gallery.ecr.aws/sumologic/kubernetes-fluentd
FROM public.ecr.aws/sumologic/kubernetes-fluentd:1.16.5-sumo-0-alpine as sumo

FROM ruby:3.2.4-alpine3.18 as base

ARG BUILD_DEPS=" \
      make gcc g++ libc-dev \
      wget bzip2 zlib-dev git linux-headers \
      automake autoconf libtool build-base \
      ruby-dev libc6-compat geoip-dev \
      snappy-dev gnupg bash \
      "

RUN addgroup -S -g 101 fluent && adduser -S -G fluent -u 100 fluent \
    # for log storage (maybe shared with host)
    && mkdir -p /fluentd/log \
    # configuration/plugins path (default: copied from .)
    && mkdir -p /fluentd/etc /fluentd/plugins \
    && chown -R fluent /fluentd && chgrp -R fluent /fluentd \
    && mkdir -p /buffers && chown -R fluent /buffers && gem sources --remove https://rubygems.org/ && gem sources -a https://mirrors.aliyun.com/rubygems/

COPY --from=sumo --chown=fluent:fluent /usr/local/bundle /usr/local/bundle

RUN sed -i 's/dl-cdn.alpinelinux.org/mirrors.aliyun.com/g' /etc/apk/repositories && apk add --no-cache \
        ca-certificates \
        ruby ruby-irb ruby-etc \
        tini libmaxminddb geoip \
        snappy \
 && apk add --no-cache $BUILD_DEPS \
 && apk add --no-cache "libssl3>=3.1.4-r1" \
 && apk add --no-cache "libcrypto3>=3.1.4-r1" \
 && echo 'gem: --no-document' >> /etc/gemrc \
 # BASEIMAGEDEPS
 # need to rebuild these gems to have the extensions installed
 && fluent-gem install \
         bigdecimal:1.4.4 \
         oj:3.16.1 \
         google-protobuf:3.21.12 \
         nio4r:2.7.3 \
         snappy:0.3.0 \
 # The only required fluentd dependency is the label router
 # kubeclient install is upgrading the one from the base image
 && fluent-gem install specific_install -v 0.3.8 \
 && fluent-gem specific_install -l https://github.com/ManageIQ/kubeclient.git --ref 054bff2c5e31a555004be2b3c4d32fb9dc5e6a0f \
 && fluent-gem specific_install -l https://github.com/kube-logging/fluent-plugin-label-router.git --ref 6d9d34a7e3fd73a400a31fb35513781802e7b36d \
 && find /usr/local/bundle/gems/ -newer /etc/gemrc -exec chown fluent:fluent {} \; \
 && rm -rf /usr/local/bundle/cache/* && find /usr/local/bundle -name '*.o' -delete

USER root
```

build image

```
docker build -t gem-build . -f Dockerfile-base
```

and then update 

```
docker run --rm -v .:/app -w /app gembuild bundle update
```